### ISSUE TITLE
It doesn't catch exit code

### ISSUE THREAD

**Issue by contributorpw** [stylemistake/runner#39]
## runnerfile

```bash
#!/usr/bin/env bash

cd "$(dirname "$0")" || exit
source ./runner

task_bar() {
    exit 0
}

```

## invocation

```sh
$> ./runnerfile bar
[10:38:51.574] Starting 'bar'...
```

There is no 'Finished' log

How can I settings this for valid fulfilled invocation?

**Comment by stylemistake** [comment]
You are exiting in the middle of the task, so that's expected. What you want is "return".

**Comment by contributorpw** [comment]
I would like to go back up the call stack. Maybe I can pass some value to the script or should I use conditional branching?

I look at the code of `runner` and I realize that I cannot interrupt the script with `exit`.

**Comment by stylemistake** [comment]
If you return a non-zero value, task will fail and bring all other dependent tasks down. You can also use `set -e`, but i'm not sure how that behaves on current master.

**Comment by contributorpw** [comment]
Thank you!

It's expected for me now.

So in general it's a good app!
